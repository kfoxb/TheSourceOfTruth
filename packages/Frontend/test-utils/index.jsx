/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    const wrapper = shallow(<Component {...props} />);
    expect(wrapper).toMatchSnapshot();
  };
