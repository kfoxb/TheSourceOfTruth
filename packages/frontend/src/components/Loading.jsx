import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import LinearProgress from '@material-ui/core/LinearProgress';
import colors from '../constants/colors';

const styles = {
  root: {
    backgroundColor: `${colors.grey}`,
    flexGrow: 1,
    left: 0,
    position: 'fixed',
    top: '40px',
    width: '100%',
  },
  barColorPrimary: {
    backgroundColor: `${colors.blue}`,
  },
};

function LinearIndeterminate(props) {
  const { classes } = props;
  return (
    <div >
      <LinearProgress
        classes={{
          root: classes.root,
          barColorPrimary: classes.barColorPrimary,
        }}
      />
    </div>
  );
}

LinearIndeterminate.propTypes = {
  classes: PropTypes.shape({
    root: PropTypes.string.isRequired,
    barColorPrimary: PropTypes.string.isRequired,
  }).isRequired,
};

export default withStyles(styles)(LinearIndeterminate);
