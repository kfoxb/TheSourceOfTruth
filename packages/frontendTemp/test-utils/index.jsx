/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const makeShallowCreateWrapper = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    return shallow(<Component {...props} />);
  };

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) => {
  const createWrapper = makeShallowCreateWrapper(Component, defaultProps);
  return (extraProps) => {
    expect(createWrapper(extraProps)).toMatchSnapshot();
  };
};
