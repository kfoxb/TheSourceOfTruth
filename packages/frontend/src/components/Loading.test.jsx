/* eslint-env jest */
import { createAssertWithPropsToMatchSnapshot } from '../../test-utils';
import Loading from './Loading';

describe('Loading', () => {
  it('should render', () => {
    createAssertWithPropsToMatchSnapshot(Loading, {})();
  });
});
