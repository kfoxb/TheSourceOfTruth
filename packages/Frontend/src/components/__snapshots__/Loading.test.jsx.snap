// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render 1`] = `
<LinearIndeterminate
  classes={
    Object {
      "barColorPrimary": "LinearIndeterminate-barColorPrimary-2",
      "root": "LinearIndeterminate-root-1",
    }
  }
/>
`;
